src/values.rs

```diff
@@ -309,14 +310,106 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
+    use std::collections::hash_map::RandomState;
+
+    macro_rules! assert_empty {
+        ($x:expr) => {
+            assert!($x.is_empty());
+            assert_eq!($x.len(), 0);
+            assert_eq!($x.iter().count(), 0);
+            assert_eq!($x.into_iter().count(), 0);
+            assert_eq!($x.get_one(), None);
+        };
+    }
+
+    macro_rules! assert_len {
```

Comment:
I could make `assert_empty` a special case of `assert_len` but I didn't think that was quite as readable. Open to other options though!

Reply:
No, that seems reasonable to me!